backtesting/backtesting.py

```diff
@@ -1446,7 +1458,10 @@ def objective_function(**params):
                     kappa=3,
                     n_initial_points=min(max_tries, 20 + 3 * len(kwargs)),
                     initial_point_generator='lhs',  # 'sobel' requires n_initial_points ~ 2**N
-                    callback=DeltaXStopper(9e-7),
+                    callback=[
+                        DeltaXStopper(9e-7),
+                        SkoptProgressBar(total=max_tries, desc="Skopt optimizations")
```

Comment:
What if instead of the callback machinery, we made a simple `progress` _variable_ and via closure call its `.update()` in `objective_function()` above (ideally at the start as skopt has its issues)? That way, we don't need a whole class for it.

Reply:
I think that should work too, going to test it. I used the same implementation as their timer class and was thinking of just switching it out when they implement an official callback. No biggie :p 

https://github.com/scikit-optimize/scikit-optimize/blob/530da127c0e3d92fc5018115585e73fecced12a5/skopt/callbacks.py#L116-L141